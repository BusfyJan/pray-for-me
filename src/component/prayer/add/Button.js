import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import MaterialButton from '@material-ui/core/Button'
import IconAdd from '@material-ui/icons/Add'

const MaterialButtonStyled = styled(MaterialButton)`
  position: fixed !important;
  right: 7px;
  bottom: 7px;
  z-index: 2;
`

const Button = ({ onClick }) => (
  <MaterialButtonStyled onClick={onClick} variant="fab" color="primary">
    <IconAdd />
  </MaterialButtonStyled>
)

Button.propTypes = {
  onClick: PropTypes.func
}

export default Button
