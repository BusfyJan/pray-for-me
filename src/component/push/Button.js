import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import MaterialButton from 'material-ui/Button'
import IconNotificationsNone from 'material-ui-icons/NotificationsNone'
import IconNotificationsActive from 'material-ui-icons/NotificationsActive'
import IconNotificationsOff from 'material-ui-icons/NotificationsOff'
import IconLoading from 'material-ui-icons/Cached'

import Rotating from 'component/util/Rotating.js'

const MaterialButtonStyled = styled(MaterialButton)`
  position: fixed !important;
  left: 7px;
  bottom: 7px;
  z-index: 2;
`

const getButtonIcon = (isWorking, isPushEnabled, canPushBeEnabled) => {
  if (isWorking) {
    return (
      <Rotating>
        <IconLoading />
      </Rotating>
    )
  }

  if (isPushEnabled) {
    return <IconNotificationsActive />
  }

  if (canPushBeEnabled) {
    return <IconNotificationsNone />
  }

  return <IconNotificationsOff />
}

const Button = ({ isWorking, isPushEnabled, canPushBeEnabled }) => (
  <MaterialButtonStyled
    onClick={() => {
      this.props.onClick()
    }}
    fab
    mini
    color="accent"
  >
    {getButtonIcon(isWorking, isPushEnabled, canPushBeEnabled)}
  </MaterialButtonStyled>
)

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  isPushEnabled: PropTypes.bool,
  canPushBeEnabled: PropTypes.bool,
  isWorking: PropTypes.bool
}

Button.defaultProps = {
  isPushEnabled: false,
  canPushBeEnabled: false,
  isWorking: false
}

export default Button
