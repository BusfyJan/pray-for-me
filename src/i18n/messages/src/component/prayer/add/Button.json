[
  {
    "id": "component.prayer.add.button.title",
    "defaultMessage": "Add new prayer request"
  }
]