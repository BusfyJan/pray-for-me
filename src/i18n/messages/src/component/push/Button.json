[
  {
    "id": "component.push.button.title",
    "defaultMessage": "Open notification settings"
  }
]