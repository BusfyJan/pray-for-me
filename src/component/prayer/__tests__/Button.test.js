import React from 'react'
import Button from '../add/Button'
import renderer from 'react-test-renderer'

const mockCallBack = jest.fn()

test('Button render', () => {
  const component = renderer.create(<Button onClick={mockCallBack} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()

  expect(mockCallBack).toBeCalled()
})
