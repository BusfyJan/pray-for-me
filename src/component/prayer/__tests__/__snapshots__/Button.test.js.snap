// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render 1`] = `
<button
  className="MuiButtonBase-root-16 MuiButton-root-1 MuiButton-raised-8 MuiButton-fab-14 MuiButton-raisedPrimary-10 sc-bdVaJa dBljYH"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label-3"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-18"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-25"
  />
</button>
`;
